sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -289,53 +289,61 @@ case class Elt(children: Seq[Expression])
     val index = indexExpr.genCode(ctx)
     val strings = stringExprs.map(_.genCode(ctx))
     val indexVal = ctx.freshName("index")
+
+    // -1 means the given index doesn't match indices of strings in split function.
+    val NOT_MATCHED = -1
+    // 0 means the given index matches one of indices of strings in split function.
```

Comment:
only 2 possible values, we can use boolean

Reply:
yea, missing it.